corehq/apps/auditcare/models.py

```diff
@@ -78,14 +79,18 @@ def description(self):
 
     class Meta:
         abstract = True
-        index_together = [("user", "event_date")]
+        index_together = [
+            ("user", "event_date"),
+            ("domain", "event_date"),
+        ]
```

Comment:
My understanding is that adding the date to the index makes it much more performant to query for a user or domain in a date range, which I believe is the primary use case for queries on auditcare data. Otherwise the query planner will evaluate any date criteria as a `FILTER` condition, which gets very slow if there is a lot of data. I do realize this adds write performance overhead, but I don't think we have a choice if we want to keep queries performant. Granted, I have not measured the performance, so all this is just theorizing at this point.

My mandate in this effort has been to move auditcare from Couch to SQL, and I have tried to preserve (or improve in the case of the domain index) existing functionality as much as possible unless it is unused. Maybe it would be worth doing a separate performance analysis?

Reply:
Possibly, but I think this might be more my unfamiliarity with multi-indices. From [Postgres docs](https://www.postgresql.org/docs/10/indexes-multicolumn.html), it seems like these will be very efficient for user or domain queries. It seems reasonable that we might want to retrieve all logs for a given date range, and then I'm not sure how performant these indices would be, but I'm just spitballing.